manifests/client.pp

```diff
@@ -77,9 +77,12 @@
       ensure => $package_ensure,
       name   => $package_name,
     }
-    $req = Package['snmp-client']
-  } else {
-    $req = undef
+  }
+
+  $req = $::osfamily ? {
+    'RedHat' => [Package['snmp-client'], Package[$snmp::params::package_name]],
```

Comment:
I think you can reference this as Package['snmpd'] since it already contains the logic that you're replicating